### PR TITLE
Added create cluster functionality for gluster

### PR DESCRIPTION
**Pull request by shtripat**
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by r0h4n** [comment]
Is there a spec for this? I need some feature name and youll need to send this PR on that feature branch on commons

**Comment by shtripat** [comment]
I have a WIP spec at https://github.com/Tendrl/specifications/pull/149

**Comment by shtripat** [comment]
@nnDarshan plz review

**Comment by shtripat** [comment]
Verified the changes with gluster and ceph cluster creation. Both work as expected.

File: tendrl/commons/flows/authorize_ssh_key/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class AuthorizeSshKey(flows.BaseFlow):
+    def run(self):
+        ssh_key = self.parameters['ssh_key']
+        ret_val, err = authorize_key.AuthorizeKey(ssh_key).run()
+        if ret_val is not True or err != "":
+            raise FlowExecutionFailedError(err)
+        return True
+
+    def load_definition(self):
```
Comment (r0h4n) [inline tendrl/commons/flows/authorize_ssh_key/__init__.py]: You cannot override load_definition(), either mark this flow with class variable "internal=True" or define this flow correctly in definitions (.yml) file

File: tendrl/commons/flows/create_cluster/__init__.py [hunk 2]
```diff
@@ -52,9 +28,12 @@ def run(self):
...
 
-        # SSH setup jobs finished above, now install ceph
-        ceph_help.create_ceph(self.parameters)
+        # SSH setup jobs finished above, now install sds bits and create cluster
+        if self.parameters['sds_type'] == "ceph":
+            ceph_help.create_ceph(self.parameters)
+        else:
+            gluster_help.create_gluster(self.parameters)
 
     def load_definition(self):
```
Comment (r0h4n) [inline tendrl/commons/flows/create_cluster/__init__.py]: You cannot override load_definition(), either mark this flow with class variable "internal=True" or define this flow correctly in definitions (.yml) file

File: tendrl/commons/flows/setup_ssh/__init__.py [hunk 3]
```diff
@@ -26,5 +28,5 @@ def run(self):
         os.remove(_temp_file.name)
 
     def load_definition(self):
```
Comment (r0h4n) [inline tendrl/commons/flows/setup_ssh/__init__.py]: You cannot override load_definition(), either mark this flow with class variable "internal=True" or define this flow correctly in definitions (.yml) file

File: tendrl/commons/flows/utils.py [hunk 4]
```diff
@@ -1,2 +1,76 @@
...
+
+from tendrl.commons.objects.job import Job
+
+LOG = logging.getLogger(__name__)
+
+
 def to_camel_case(snake_str):
     return "".join(x.title() for x in snake_str.split('_'))
+
+def ceph_setup_ssh(parameters):
```
Comment (r0h4n) [inline tendrl/commons/flows/utils.py]: call this ceph_create_ssh_setup_jobs and likewise for gluster

File: tendrl/commons/flows/authorize_ssh_key/__init__.py (hunk 1 above)
Comment (shtripat) [inline tendrl/commons/flows/authorize_ssh_key/__init__.py]: Done

File: tendrl/commons/flows/create_cluster/__init__.py (hunk 2 above)
Comment (shtripat) [inline tendrl/commons/flows/create_cluster/__init__.py]: Done

File: tendrl/commons/flows/setup_ssh/__init__.py (hunk 3 above)
Comment (shtripat) [inline tendrl/commons/flows/setup_ssh/__init__.py]: Done

File: tendrl/commons/flows/utils.py (hunk 4 above)
Comment (shtripat) [inline tendrl/commons/flows/utils.py]: Ok. will change it

**Pull request merged**